### PR TITLE
Make "runtime tokens" parameter optional

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

**Comment by fbiville** [comment]
The CI currently fails against the Dataflow server, but not the direct runner (`mvn clean verify -am -pl v2/googlecloud-to-neo4j` locally runs file). 
Is there something missing in terms of template metadata update?

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java [hunk 1]
```diff
@@ -50,6 +50,7 @@ public interface Neo4jFlexTemplateOptions extends CommonTemplateOptions {
       description = "Options JSON",
```
Comment (bvolpato) [inline v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/options/Neo4jFlexTemplateOptions.java:50]: `optional = true`

**Comment by bvolpato** [comment]
I think mandatory parameters (@TemplateParameter `optional` tag) are only checked on Dataflow. It's a gap that we'll have to cover on our direct client. But if you run without direct runner (`-PtemplatesIntegrationTests`) you should see it.

**Comment by bvolpato** [comment]
It seems counterintuitive because you have the `@Default` -- but you still have to tell Dataflow what are the mandatory parameters, because it validates requests based on them, without having access to the class / JAR. Perhaps we could use `@Default` to imply `optional = true`, but I'd have to give it a look for potential impact.

One concern that I have today is related to the order that things show up in the UI (this shouldn't be a blocker, but may cause somewhat bad UX).

**Review by bvolpato — APPROVED** [review]

**Pull request merged**